### ISSUE TITLE
Refactor com.sun.* classes to org.glassfish.main

### ISSUE THREAD

**Issue by arjantijms** [eclipse-ee4j/glassfish#23627]
GlassFish uses a mix of `org.glassfish` and `com.sun.*` packages. For consistency, and because Eclipse doesn't own `com.sun` or the domain `sun.com` it may be better to refactor everything to `org.glassfish.*`. This may however be quite an operation.

There's also a bunch of technically internal classes exposed in things like `domain.xml` starting with `com.sun`.

**Comment by dmatej** [comment]
I agree, but it should be done in some iterative process, because there is pretty much of "magical" reflection. In each iteration we need to pass all available tests (including those obsoleted, TCK, javaee-samples, ... I burned my fingers just with changing names of generated interfaces.

**Comment by dmatej** [comment]
I think that the time has come

**Comment by arjantijms** [comment]
`org.glassfish.main` probably. Not that it's such a good name, but all the other packages use it.

**Comment by arjantijms** [comment]
We may still want to do this. One day.

**Comment by renatsaf** [comment]
Hi @dmatej I asked Claude to analyze the issue.

**Comment by renatsaf** [comment]
Following up on my analysis of the current state of this refactor.

**Scope in the current codebase:**

| | Count |
|---|---|
| `.java` files declaring `package com.sun.*` | ~5,171 |
| — legacy test/sample packages (`com.sun.s1asdev` / `com.sun.s1peqe`) | ~1,734 |
| — production/core `com.sun.*` files | ~3,400 |
| Files referencing `com.sun` in `*.xml` | ~974 |

Largest production clusters: `com.sun.enterprise.deployment`, `com.sun.enterprise.config.serverbeans`, `com.sun.ejb.containers`, `com.sun.jts.CosTransactions`, `com.sun.enterprise.web`, `com.sun.enterprise.security.*`.

**Why this can't be a mechanical IDE rename** (echoing @dmatej's "magical reflection" warning):

1. **Class names appear as string literals** in `domain.xml`, login configs and templates, so they're invisible to refactoring tools and must be changed in lockstep — e.g. `com.sun.enterprise.config.serverbeans.AppserverConfigEnvironmentFactory`, `com.sun.enterprise.naming.SerialInitContextFactory`, `com.sun.enterprise.security.auth.login.FileLoginModule`, `...realm.file.FileRealm`, `...security.jacc.provider.Simple*`, `...security.ssl.GlassfishSSLImpl`.
2. **External projects bind to our `com.sun` names by reflection** — notably EclipseLink's `GlassFishPlatform` unwrapping `com.sun.gjc.spi.base.ConnectionHolder` (already noted in the issue), plus TCK / javaee-samples.
3. **Must be left alone** (not GlassFish-owned, or a separate compatibility decision): `com.sun.aas.*` system properties, and third-party packages `com.sun.faces.*`, `com.sun.corba.ee.*`, `com.sun.xml.ws.*` / `com.sun.xml.wss.*`, `com.sun.jndi.ldap`, `com.sun.sql.jdbcx.*`.

**Proposed approach** (matching the iterative process @dmatej asked for, target root `org.glassfish.main` per @arjantijms):
- One PR per cohesive leaf package; never big-bang.
- For each: rename package -> fix Java imports -> sweep XML / `.policy` / `login.conf` / `META-INF/services` / templates for the old FQCN as a string -> run the module's tests + smoke before merging.
- Keep externally-contracted names (EclipseLink `ConnectionHolder`, system properties) frozen until coordinated downstream.
- Defer the ~1,734 `s1asdev`/`s1peqe` legacy test/sample packages (low value, high churn) — possibly out of scope entirely.

Happy to start with a single low-risk module (one with no XML/reflection exposure and few reverse-dependencies) to prove the workflow if that sounds good.

**Comment by renatsaf** [comment]
As a concrete first step I opened #26102, which renames `com.sun.enterprise.backup.util.BackupUtils` to `org.glassfish.main.enterprise.backup.util.BackupUtils`.

I picked this class deliberately as the lowest-risk starting point to validate the per-package workflow (rename -> fix imports -> sweep config/string references -> compile): it's a single class contained in the `appserver/admin/backup` module, referenced from only two other in-module files, with no `domain.xml`/config/`Class.forName` exposure and no external-project contract on it.

If the approach in #26102 looks right, I'll continue iteratively, one leaf package per PR, keeping each step green — starting with the other reflection-free packages before tackling the config/TCK-exposed ones.
